### DISCUSSION TITLE
[Discussion] Log API

### DISCUSSION THREAD

**Discussion by haoadoreorange** [niieani/bash-oo-framework#95] [Ideas]
Hello,

I find the current Logging API a bit unsual in the sense that I need to add a specific Logger to get the context when logging. What if I want most of the time using `INFO` and on some line `DEBUG`.
Normally in other logging library it is something like `Log::DEBUG` or `Log::INFO`.

I see that there alreays is `Log::Using`, but it doesn't get any context with that. May be it should use the default namespace ?

**Comment by niieani** [comment]
Thanks @haoadoresorange, we can definitely take this into account for the [3.0](https://github.com/niieani/bash-oo-framework/discussions/91). Unfortunately the project is a little stalled these days as I have other priorities. But maybe one of the days I'll come back.
